plugins/traction_innkeeper/traction_innkeeper/v1_0/tenant/routes.py

```diff
@@ -0,0 +1,223 @@
+
+
+@docs(
+    tags=[SWAGGER_CATEGORY],
+)
+@response_schema(TenantRecordSchema(), 200, description="")
+@error_handler
+async def tenant_self(request: web.BaseRequest):
+    context: AdminRequestContext = request["context"]
+    # we need the caller's wallet id
+    wallet_id = context.profile.settings.get("wallet.id")
+
+    # records are under base/root profile, use Tenant Manager profile
+    mgr = context.inject(TenantManager)
+    profile = mgr.profile
+
+    async with profile.session() as session:
+        # tenant's must always fetch by their wallet id.
+        rec = await TenantRecord.query_by_wallet_id(session, wallet_id)
+        LOGGER.info(rec)
+
+    return web.json_response(rec.serialize())
+
+
+@docs(tags=[SWAGGER_CATEGORY], summary="Register a Public DID")
+@querystring_schema(RegisterPublicDidSchema())
+@response_schema(schema=RegisterPublicDidSchema, code=200, description="")
+@error_handler
+async def register_public_did(request: web.BaseRequest):
+    LOGGER.info("> tenant_register_public_did()")
+    context: AdminRequestContext = request["context"]
+
+    # check if there is a public did
+    info = None
+    async with context.session() as session:
+        wallet = session.inject_or(BaseWallet)
+        if not wallet:
+            raise web.HTTPForbidden(reason="No wallet available")
+        try:
+            info = await wallet.get_public_did()
+        except WalletError as err:
+            raise web.HTTPBadRequest(reason=err.roll_up) from err
+    if info:
+        raise web.HTTPConflict(reason=f"Wallet already has a public DID: `{info.did}`")
+
+    did = request.query.get("did")
+    verkey = request.query.get("verkey")
+    if not did or not verkey:
+        raise web.HTTPBadRequest(
+            reason="Request query must include both did and verkey"
+        )
+
+    alias = request.query.get("alias")
+    if not alias:
+        alias = context.profile.settings.get("wallet.id")
+
+    genesis_url = context.profile.settings.get("ledger.genesis_url")
+    did_registration_url = genesis_url.replace("genesis", "register")
+    data = {
+        "did": did,
+        "verkey": verkey,
+        "alias": alias,
+    }
+    requests.post(did_registration_url, json=data)
+    LOGGER.info(f"did_registration_url = {did_registration_url}, data = {data}")
+
+    LOGGER.info("< tenant_register_public_did()")
+    return web.json_response(data)
+
+
+@docs(tags=[SWAGGER_CATEGORY], summary="Get a tenant subwallet")
+@response_schema(WalletRecordSchema(), 200, description="")
+@error_handler
+async def tenant_wallet_get(request: web.BaseRequest):
+    context: AdminRequestContext = request["context"]
+    # we need the caller's wallet id
+    wallet_id = context.profile.settings.get("wallet.id")
+
+    # records are under base/root profile, use Tenant Manager profile
+    mgr = context.inject(TenantManager)
+    profile = mgr.profile
+
+    # this is from multitenant / admin / routes.py -> wallet_get
+    # duplicate code.
```

Comment:
Would it make sense (and would it be possible), if the code is the same, to just trigger the original handler and return its result, rather than duplicating the code here?

Reply:
don't think it's possible as you will need to manipulate the request itself to get into the call...(need to put the wallet_id into the request URL) could redirect? but that seems kind of weird too.